app/uk/gov/hmrc/digitalservicestax/views/uniform/StringField.scala.html

```diff
@@ -0,0 +1,134 @@
+@*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *@
+
+@import cats.implicits._
+@import play.twirl.api.HtmlFormat
+
+@this(
+        standardField: StandardField,
+        govukDetails: GovukDetails,
+        govukInput: GovukInput
+)
+
+@(key: List[String], values: String, errors: ErrorTree, messages: UniformMessages[Html], autoFields: Option[String] = None, inputClass: String = "govuk-!-width-two-thirds")
+
+@keyWithDots=@{key.mkString(".")}
+@headKey=@{key.headOption.getOrElse("")}
+
+@percentageInput=@{
+    headKey === "report-social-media-operating-margin" ||
+    headKey === "report-search-engine-operating-margin" ||
+    headKey === "report-online-marketplace-operating-margin"
+}
+
+@currencyInput=@{
+    headKey === "relief-deducted"||
+    headKey === "allowance-deducted"||
+    headKey === "group-liability"||
+    headKey.contains("company-liabilities")
+}
+
+@telInput=@{key.contains("phoneNumber")}
+
+@autoComplete=@{key match {
+    case k if k.contains("forename") => "given-name"
+    case k if k.contains("surname") => "family-name"
+    case k if k.contains("phoneNumber") => "tel"
+    case k if k.contains("line1") => "address-line1"
+    case k if k.contains("line2") => "address-line2"
+    case k if k.contains("line3") => "address-line3"
+    case k if k.contains("line4") => "address-level4"
```

Comment:
should that be line4?

Reply:
I'll double check this but the "address-level4" string is used elsewhere so assuming correct for now.